### PR TITLE
CRUD email forwarding commands

### PR DESCRIPTION
**Pull request by sapirshuker**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [] In Progress
- [X] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [CIAC-1819](https://jira-hq.paloaltonetworks.local/browse/CIAC-1819)

## Description
add CRUD command for gmail for email forwarding

## Minimum version of Cortex XSOAR
- [X] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [X] No

## Must have
- [X] Tests
- [X] Documentation

### CODE REVIEW

**Comment by ShirleyDenkberg** [comment]
@Shellyber Doc review completed.

File: Packs/Gmail/Integrations/Gmail/README.md [hunk 1]
```diff
@@ -5044,3 +5004,121 @@ This command is only available to service account clients who have been delegate
...
+***
+Creates a forwarding address. If ownership verification is required, a message will be sent to the recipient and the resource's verification status will be set to pending; otherwise, the resource will be created with verification status set to accepted. This method is only available to service account clients that have been delegated domain-wide authority. The special value "me" can be used to indicate the authenticated user.
+
+
+#### Base Command
+
+`gmail-forwarding-address-add`
+#### Input
+
+| **Argument Name** | **Description** | **Required** |
```
Comment (Shellyber) [inline Packs/Gmail/Integrations/Gmail/README.md:5018]: Need to update the readme with the new changes and remove what is not necessary (like the disposition here)

File: Packs/Gmail/Integrations/Gmail/test_data/input_data.py [hunk 2]
```diff
@@ -270,3 +270,16 @@
                                          'internalDate': '1665491175000'}]
 expected_result_test_emails_to_entry = {"expected_human_readable": expected_human_readable_test_emails_to_entry,
                                         "except_contents": except_contents_test_emails_to_entry}
+
+# test_forwarding_address_get_command #
+expected_result_forwarding_address_get_command_1 = {"raw_response": {'forwardingEmail': 'test@gmail.com',
```
Comment (Shellyber) [inline Packs/Gmail/Integrations/Gmail/test_data/input_data.py:275]: Let's move this to the test command

**Review by Shellyber — CHANGES_REQUESTED** [review]
Nice!

- Please review again all the docstrings - the descriptions of the arguments seems wrong in some of the commands
- Initialize the variables with the right object instead of None
- Give examples of the new commands in the README

File: Packs/Gmail/Integrations/Gmail/Gmail.py [hunk 3]
```diff
@@ -2058,6 +2180,128 @@ def send_as_add_command():
...
+            user_id: str - The user's email address or the user id.
+            limit: str - The Limit of the results list. Default is 50.
+        Returns:
+            A CommandResults object.
+    """
+    args = demisto.args()
+    user_id = args.get('user_id')
+    limit = int(args.get('limit', '50'))
+    result = forwarding_address_list(user_id)
+    context = result.get('forwardingAddresses')[:limit]  # type: ignore
```
[the remark this answers is not in the document]
Comment (sapirshuker) [reply Packs/Gmail/Integrations/Gmail/Gmail.py:2292]: "message": "Value of type \"Optional[Any]\" is not indexable".
even if i change it to 
context = result.get('forwardingAddresses')[:limit] if result else None

[the remark this answers is not in the document]
Comment (Shellyber) [reply Packs/Gmail/Integrations/Gmail/Gmail.py]: Nope, this is why I asked. We need to handle this better. 
Don't do the limit here. 
Check if context in the command results than limit it if not empty

File: Packs/Gmail/Integrations/Gmail/Gmail.py [hunk 4]
```diff
@@ -1979,15 +1981,16 @@ def reply_mail_command():
     return sent_mail_to_entry('Email sent:', [result], emailto, emailfrom, cc, bcc, body, subject)
 
 
-def forwarding_address_add(user_id: str, forwarding_email: str) -> tuple:
+def forwarding_address_add(user_id: str, forwarding_email: str) -> tuple[dict, bool, Optional[dict]]:
     """ Creates forwarding address.
         Args:
-            user_id: str - The repository the user entered, if he did.
+            user_id: str - The user's email address or the user id.
             forwarding_email: str - The name of the file
```
Comment (Shellyber) [inline Packs/Gmail/Integrations/Gmail/Gmail.py:1988]: This is not the name of the file - please have a look again

File: Packs/Gmail/Integrations/Gmail/Gmail.py [hunk 5]
```diff
@@ -1979,15 +1981,16 @@ def reply_mail_command():
...
         Returns:
             A Response object.
     """
-    result = None
+    result = {}
+    exception_details = None
```
Comment (Shellyber) [inline Packs/Gmail/Integrations/Gmail/Gmail.py:1993]: exception_details = {}

File: Packs/Gmail/Integrations/Gmail/Gmail.py [hunk 6]
```diff
@@ -2002,14 +2005,14 @@ def forwarding_address_add(user_id: str, forwarding_email: str) -> tuple:
...
-    return result, exception, {'forwardingEmail': forwarding_email, 'errorMessage': error_message, 'userId': user_id}
+        exception_details = {'forwardingEmail': forwarding_email, 'errorMessage': error_message, 'userId': user_id}
+    return result, exception, exception_details
 
 
 def forwarding_address_add_command() -> list[CommandResults]:
     """ Creates forwarding address.
         Args:
-            user_id: str - The repository the user entered, if he did.
+            user_id: str - The user's email address or the user id.
```
Comment (Shellyber) [inline Packs/Gmail/Integrations/Gmail/Gmail.py:2015]: same error in the description of the email

File: Packs/Gmail/Integrations/Gmail/Gmail.py [hunk 7]
```diff
@@ -2057,8 +2060,9 @@ def forwarding_address_update(user_id: str, disposition: str, forwarding_email:
         Returns:
             A dict object.
     """
-    exception = None
-    result = None
+    exception = False
+    result = {}
+    exception_details = None
```
Comment (Shellyber) [inline Packs/Gmail/Integrations/Gmail/Gmail.py:2065]: exception_details = {}

File: Packs/Gmail/Integrations/Gmail/Gmail.py [hunk 8]
```diff
@@ -2289,7 +2295,7 @@ def forwarding_address_list_command() -> CommandResults:
     user_id = args.get('user_id')
     limit = int(args.get('limit', '50'))
     result = forwarding_address_list(user_id)
-    context = result.get('forwardingAddresses')[:limit]  # type: ignore
+    context = result.get('forwardingAddresses')[:limit] if result else None  # type: ignore
```
Comment (Shellyber) [inline Packs/Gmail/Integrations/Gmail/Gmail.py:2298]: Remove the type ignore, and deal with the situation where it is empty

File: Packs/Gmail/Integrations/Gmail/Gmail.yml [hunk 9]
```diff
@@ -1988,12 +1978,6 @@ script:
     - contextPath: Gmail.ForwardingAddress.verificationStatus
       description: Indicates whether this address has been verified and is usable for forwarding.
       type: String
-    - contextPath: Gmail.ForwardingAddress.Disposition
-      description: The state that a message should be left in after it has been forwarded.
-      type: String
-    - contextPath: Gmail.ForwardingAddress.Enabled
```
Comment (Shellyber) [inline Packs/Gmail/Integrations/Gmail/Gmail.yml]: The enabled is part of the disposition? not the add?

File: Packs/Gmail/Integrations/Gmail/Gmail.yml [hunk 10]
```diff
@@ -2203,12 +2187,6 @@ script:
     - contextPath: Gmail.ForwardingAddress.forwardingEmail
       description: An email address to which messages can be forwarded.
       type: String
-    - contextPath: Gmail.ForwardingAddress.Disposition
-      description: The state that a message should be left in after it has been forwarded.
-      type: String
-    - contextPath: Gmail.ForwardingAddress.Enabled
```
Comment (Shellyber) [inline Packs/Gmail/Integrations/Gmail/Gmail.yml]: same question

**Review by Shellyber — CHANGES_REQUESTED** [review]
The README is still missing the command example (not just how to run in on the CLI)
Need to also add the context example and human-readable outputs 
Use the demisto-sdk generate-docs command to do it

File: Packs/Gmail/Integrations/Gmail/Gmail.yml (hunk 9 above)
Comment (sapirshuker) [reply Packs/Gmail/Integrations/Gmail/Gmail.yml]: Yes, it part of update, attached API doc: 
https://developers.google.com/gmail/api/reference/rest/v1/AutoForwarding

File: Packs/Gmail/Integrations/Gmail/Gmail.yml (hunk 10 above)
Comment (sapirshuker) [reply Packs/Gmail/Integrations/Gmail/Gmail.yml]: there is only one contextPath of Gmail.ForwardingAddress.Enabled in the update command

File: Packs/Gmail/Integrations/Gmail/README.md [hunk 11]
```diff
@@ -147,7 +147,10 @@ After you successfully execute a command, a DBot message appears in the War Room
 - Reply an email using Gmail: **reply-mail**
 - Removers a delegate from a mailbox: **gmail-remove-delegated-mailbox**
 - Get details of a specific role: **gmail-get-role**
-- Creates a forwarding address.: **gmail-forwarding-address-add**
+- Creates a forwarding address: **gmail-forwarding-address-add**
+- Get the specified forwarding address or a list of forwarding addresses for the specified account: **gmail-forwarding-address-get**
```
[the remark this answers is not in the document]
Comment (Shellyber) [reply Packs/Gmail/Integrations/Gmail/README.md:151]: What about this?

File: Packs/Gmail/Integrations/Gmail/Gmail.py [hunk 12]
```diff
@@ -1976,32 +1981,158 @@ def reply_mail_command():
...
+            disposition: str - The state that a message should be left in after it has been forwarded.
+        Returns:
+            result: dict - Response body from the API.
+            exception: bool - Indicates whether there is an error.
+            exception_details: dict - The details of the exception.
+    """
+    exception = False
+    result = {}
+    exception_details = {}
+    error_message = ''
```
Comment (Shellyber) [inline Packs/Gmail/Integrations/Gmail/Gmail.py:2070]: we don't need this

**Review by Shellyber — CHANGES_REQUESTED** [review]
You missed my general note before - need to fix the readme file

**Review by Shellyber — APPROVED** [review]
Great Work

**Pull request merged**